x-pack/plugins/lens/public/xy_visualization_plugin/xy_config_panel.tsx

```diff
@@ -148,15 +141,13 @@ export const XYConfigPanel = injectI18n((props: VisualizationProps<State> & Inje
 
       {state.legend.isVisible && (
         <EuiFormRow
-          label={intl.formatMessage({
+          label={i18n.translate('xpack.lens.xyChart.legendPositionLabel', {
             defaultMessage: 'Legend position',
-            id: 'xpack.lens.xyChart.legendPositionLabel',
           })}
         >
           <EuiButtonGroup
-            legend={intl.formatMessage({
+            legend={i18n.translate('xpack.lens.xyChart.legendPositionLegend', {
```

Comment:
I think the key you've chosen here is confusing, shouldn't it be `legendPositionLabel`?

Also, why are you using `label` on all of these keys?

Reply:
Hm. It should be "ButtonLabel" probably. There's already a `legendPositionLabel`. As for why I'm using "label", it's the naming convention in the guidelines: `packages/kbn-i18n/GUIDELINE.md`